cloud/aws/services/ec2/bastion.go

```diff
@@ -103,14 +103,27 @@ func (s *Service) describeBastionInstance(clusterName string, status *v1alpha1.A
 		return nil, errors.Wrap(err, "failed to describe bastion host")
 	}
 
-	if len(out.Reservations) == 0 || len(out.Reservations[0].Instances) == 0 {
-		return nil, NewNotFound(errors.New("bastion host not found"))
+	// TODO: properly handle multiple bastions found rather than just returning
+	// the first non-terminated.
```

Comment:
I think we might be able to solve this by filtering on running instances only?

Reply:
Maybe, we also need to consider that during reconciliation we might need to start stopped instances, etc.

Reply:
Sounds good, I think the TODO captures that.